src/App.js

```diff
@@ -1,14 +1,78 @@
-import './App.css';
-import Education from './components/Education';
-import Experience from './components/Experience';
-import General from './components/General';
+import { useState } from 'react';
+import styles from './app.module.css';
+import { Education, Experience, General } from './components/';
+
+const initialData = {
+  general: { name: 'John', email: 'john@john.john', phone: '+334455667788' },
+  education: [
+    {
+      school: 'Very Nice school',
+      title: 'Pupil',
+      date: { from: '2000-06', to: '2003-07' },
+    },
+    {
+      school: 'University',
+      title: 'Student',
+      date: { from: '2004-06', to: '2009-07' },
+    },
+  ],
+  experience: [
+    {
+      company: 'BBBB',
+      position: 'Worker',
+      description: 'nice',
+      date: { from: '2010-01', to: '2015-02' },
+    },
+  ],
+};
+
 function App() {
+  const [data, setData] = useState(initialData);
+
+  function handleDataChange(section, dataSection) {
+    setData({
+      ...data,
+      [section]: dataSection,
+    });
+  }
+
+  function handleAddJob() {
+    setData({
+      ...data,
+      experience: [
+        ...data.experience,
+        {
+          company: '',
+          position: '',
+          description: '',
+          date: { from: '2022-01', to: '2022-01' },
```

Comment:
use dayjs lib for date management. It's a great low cost library that has everything you need. You would do something like `dayjs('2022-01').toString('YYYY-MM');` here. Also, it really helps to save date as Date, be it new Date() or dayjs().toDate();. And then only parse it when you display it. Since you would send the actual Date type into backend.

Reply:
https://day.js.org/ here's the docs